File: fixtures/objc/commoncrypto/hmac-sha256/main.m
#import <Foundation/Foundation.h>
#import <CommonCrypto/CommonCrypto.h>

int main() {
    @autoreleasepool {
        NSData *key = [@"secret_key" dataUsingEncoding:NSUTF8StringEncoding];
        NSData *message = [@"Hello, World!" dataUsingEncoding:NSUTF8StringEncoding];
        uint8_t mac[CC_SHA256_DIGEST_LENGTH];
        
        CCHmac(kCCHmacAlgSHA256, key.bytes, key.length, 
               message.bytes, message.length, mac);
    }
    return 0;
}

File: fixtures/swift/commoncrypto/hmac-sha256/main.swift
import Foundation
import CommonCrypto

let key = "secret_key".data(using: .utf8)!
let message = "Hello, World!".data(using: .utf8)!
var mac = [UInt8](repeating: 0, count: Int(CC_SHA256_DIGEST_LENGTH))

key.withUnsafeBytes { keyBytes in
    message.withUnsafeBytes { messageBytes in
        CCHmac(CCHmacAlgorithm(kCCHmacAlgSHA256),
               keyBytes.baseAddress, key.count,
               messageBytes.baseAddress, message.count,
               &mac)
    }
}
